zebrad/src/commands/start.rs

```diff
@@ -48,15 +62,55 @@ impl StartCmd {
             }),
             1,
         );
-        let config = app_config();
-        let state = zebra_state::on_disk::init(config.state.clone());
         let (peer_set, _address_book) = zebra_network::init(config.network.clone(), node).await;
-        let verifier = zebra_consensus::chain::init(config.network.network, state.clone());
 
         let mut syncer = sync::Syncer::new(config.network.network, peer_set, state, verifier);
 
         syncer.sync().await
     }
+
+    /// Get the initial tip block, using `state`.
+    ///
+    /// If there is no tip, returns `Ok(None)`.
+    async fn initial_tip<S>(&self, state: S) -> Result<Option<Arc<Block>>, Report>
```

Comment:
I might be missing it but this doesn't seem to be accessing any state of the `StartCmd`. I think it might make more sense to make this a method on the `Verifier` and have it grab the tip from the provided state as part of its constructor, rather than passing it in as an arg.

Reply:
How can I execute async functions in a non-async constructor?

Reply:
oh yea, I'd see if you can make the constructor async

Reply:
There should be a clippy lint for async, template functions that don't actually use self.

Reply:
That function doesn't actually depend on any verifier state either, so I moved it to zebra-state in 6c987e036e88ec34bb7c03f2275eae1f2fa5ee9a.

I imagine you'll want to rewrite it as part of the chain reorg changes.